x-pack/solutions/observability/plugins/observability/public/pages/util/synthetics_alert_params_to_kql.ts

```diff
@@ -0,0 +1,120 @@
+import type { SyntheticsMonitorStatusRuleParams } from '@kbn/response-ops-rule-params/synthetics_monitor_status';
+import type { TLSRuleParams } from '@kbn/response-ops-rule-params/synthetics_tls';
+
+const MS_TO_DAYS = 1000 * 60 * 60 * 24;
+
+const supportedAlertParams = [
+  'tags',
+  'locations',
+  'monitorIds',
+  'monitorTypes',
+  'projects',
+] as const;
+
+type SupportedAlertParam = (typeof supportedAlertParams)[number];
+
+const fieldMap: Record<SupportedAlertParam, string> = {
+  locations: 'observer.name',
+  monitorIds: 'monitor.id',
+  monitorTypes: 'monitor.type',
+  projects: 'project.id',
+  tags: 'tags',
+};
+function getFieldnameForKey(key: SupportedAlertParam): string {
+  if (!fieldMap[key]) {
+    throw new Error(`Unsupported alert param: ${key}`);
+  }
+  return fieldMap[key];
+}
+
+function listToKqlNode(fieldname: string, list: string[]): KueryNode {
+  if (list.length === 0) {
+    throw new Error(`Alert params ${fieldname ?? 'field'} must be a non-empty array`);
+  }
+  return nodeBuilder.or(list.map((item) => nodeBuilder.is(fieldname, item)));
+}
+
+function mapExtraSyntheticsFilters(
+  extraFilters: Partial<Record<SupportedAlertParam, string[] | undefined>>,
+  kqlQuery?: string
+): KueryNode[] {
+  const filters: KueryNode[] = [];
+
+  if (kqlQuery) {
+    filters.push(fromKueryExpression(kqlQuery));
+  }
+
+  filters.push(
+    ...supportedAlertParams
+      .filter((key) => Array.isArray(extraFilters[key]) && extraFilters[key]!.length > 0)
+      .map((key) => listToKqlNode(getFieldnameForKey(key), extraFilters[key]!))
+  );
+
+  return filters;
+}
+
+/**
+ *
+ * @param params from a Synthetics Monitor Status Alert
+ * @returns KQL query string
+ */
+export function syntheticsMonitorStatusAlertParamsToKqlQuery(
+  params: SyntheticsMonitorStatusRuleParams
+): string {
+  const { condition, kqlQuery, ...rest } = params;
+
+  const filters = mapExtraSyntheticsFilters(rest, kqlQuery);
+
+  filters.push(nodeBuilder.is('monitor.status', 'down'));
+
+  return toKqlExpression(nodeBuilder.and(filters));
+}
+
+/**
+ * Maps the params for a Synthetics TLS Alert to a KQL query string.
+ *
+ * @param params from a Synthetics TLS Alert
+ * @returns KQL query string
+ */
+export function syntheticsTlsAlertParamsToKqlQuery(params: TLSRuleParams): string {
+  const { certAgeThreshold, certExpirationThreshold, search, kqlQuery, ...rest } = params;
+
+  const expirationFilters: KueryNode[] = [];
+  const filters: KueryNode[] = [];
+
+  if (certExpirationThreshold) {
+    expirationFilters.push(
+      nodeBuilder.range(
+        'tls.server.x509.not_after',
+        'lt',
+        new Date(Date.now() + certExpirationThreshold * MS_TO_DAYS).toISOString()
+      )
+    );
+  }
+
+  if (certAgeThreshold) {
+    expirationFilters.push(
+      nodeBuilder.range(
+        'tls.server.x509.not_before',
+        'lt',
+        new Date(Date.now() - certAgeThreshold * MS_TO_DAYS).toISOString()
```

Comment:
nit:
```suggestion
        moment().add(certExpirationThreshold, 'days').toISOString()
```

Reply:
This is definitely more readable. I was attempting to avoid referencing moment, this results in the same string, but I see we just merged similar code that references moment, so I can adjust so the code will match.

Will need to revise tests as well but that is OK.

Reply:
cf76d301e19e83d9f3c9811c224d23f31c73ba3f